### PR TITLE
User-friendly error for invalid envs/flags

### LINKED ISSUE #3407: handle empty string env vars more gracefully
## Describe the bug
When certain environment variables are set to an empty string (or something that's not parseable as a boolean), Terragrunt raises a message about `strconv.ParseBool` failing.

## Steps To Reproduce
1. Set `TERRAGRUNT_FORWARD_TF_STDOUT=""` or `TERRAGRUNT_DISABLE_LOG_FORMATTING=""`
1. Run any terragrunt command

## Expected behavior
Terragrunt either considers empty string the same as unset, or considers the variable set. IMO the former is more intuitive, but I can understand why Gruntworks might not like to do it that way; it seems also valid to only allow "true"  / "false" etc, just maybe should throw a more user-friendly error message, like
```
Invalid value for TERRAGRUNT_DISABLE_LOG_FORMATTING; must be one of true / false.
```

## Nice to haves
- [x] Terminal output
```shell
% export TERRAGRUNT_FORWARD_TF_STDOUT=""
% tg plan                               
11:39:22.441 ERROR  error parse: strconv.ParseBool: parsing "": invalid syntax
11:39:22.441 ERROR  Unable to determine underlying exit code, so Terragrunt will exit with error code 1
% export TERRAGRUNT_DISABLE_LOG_FORMATTING=oogabooga
% tg plan                                           
11:42:36.429 ERROR  error parse: strconv.ParseBool: parsing "oogabooga": invalid syntax
11:42:36.429 ERROR  Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```

## Versions

- Terragrunt version: `terragrunt version 0.67.5`
- OpenTofu/Terraform version: `OpenTofu v1.8.2`
- Environment details (Ubuntu 20.04, Windows 10, etc.): MacOS

## Additional context
Unsetting the var completely does resolve the issue
#3367

### PR DESCRIPTION
**Pull request by levkohimins**
## Description

Fixes #3407.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

File: pkg/cli/generic_flag.go [hunk 1]
```diff
@@ -199,31 +210,31 @@ func (val *genericType[T]) Set(str string) error {
...
+			return errors.Errorf(`must be one of: "0", "1", "f", "t", "false", "true"`)
 		}
 
 		*dest = v
 
 	case *int:
 		v, err := strconv.ParseInt(str, 0, strconv.IntSize)
 		if err != nil {
-			return errors.Errorf("error parse: %w", err)
+			return errors.Errorf("must be 32-bit integer")
```
Comment (yhakbar) [inline pkg/cli/generic_flag.go:221]: Why aren't you wrapping the error? It's generally a good practice to.

File: pkg/cli/generic_flag.go [hunk 2]
```diff
@@ -199,31 +210,31 @@ func (val *genericType[T]) Set(str string) error {
...
 		}
 
 		*dest = int(v)
 
 	case *uint:
 		v, err := strconv.ParseUint(str, 10, 64)
 		if err != nil {
-			return errors.Errorf("error parse: %w", err)
+			return errors.Errorf("must be 32-bit unsigned integer")
```
Comment (yhakbar) [inline pkg/cli/generic_flag.go:229]: Why aren't you wrapping the error? It's generally a good practice to.

File: pkg/cli/generic_flag.go [hunk 3]
```diff
@@ -199,31 +210,31 @@ func (val *genericType[T]) Set(str string) error {
...
 		}
 
 		*dest = uint(v)
 
 	case *int64:
 		v, err := strconv.ParseInt(str, 0, 64)
 		if err != nil {
-			return errors.Errorf("error parse: %w", err)
+			return errors.Errorf("must be 64-bit integer")
```
Comment (yhakbar) [inline pkg/cli/generic_flag.go:237]: Why aren't you wrapping the error? It's generally a good practice to.

File: pkg/cli/generic_flag.go (hunk 1 above)
Comment (levkohimins) [reply pkg/cli/generic_flag.go:221]: The issue was to clarify the error message the user gets when he specify an invalid value. 
Was:

invalid value "ddddd" for TERRAGRUNT_PARALLELISM: error parse: strconv.ParseInt: parsing "ddddd": invalid syntax


Now: 
```
 invalid value "ddddd" for TERRAGRUNT_PARALLELISM: must be 32-bit integer
```

File: pkg/cli/generic_flag.go (hunk 2 above)
Comment (levkohimins) [reply pkg/cli/generic_flag.go:229]: Explained above.

File: pkg/cli/generic_flag.go (hunk 3 above)
Comment (levkohimins) [reply pkg/cli/generic_flag.go:237]: Explained above.

File: pkg/cli/generic_flag.go (hunk 1 above)
Comment (yhakbar) [reply pkg/cli/generic_flag.go:221]: I'll approve it, deferring to your judgment, but I feel like we should be able to wrap the invalid syntax error while updating the error message.

**Review by yhakbar — DISMISSED** [review]

File: pkg/cli/generic_flag.go (hunk 1 above)
Comment (levkohimins) [reply pkg/cli/generic_flag.go]: You're right! Update the code.

Comment (levkohimins) [reply pkg/cli/generic_flag.go]: You're right! Update the code.

**Review by yhakbar — APPROVED** [review]
I'm not good at Golang. This will result in [errors.Is](https://pkg.go.dev/errors#Is) returning true, right?

**Comment by levkohimins** [comment]
> I'm not good at Golang. This will result in [errors.Is](https://pkg.go.dev/errors#Is) returning true, right?

Exactly, that's why I added `Unwrap() error`

https://github.com/gruntwork-io/terragrunt/blob/cb32d540521ca681b28f62852a5f86afade55161/pkg/cli/errors.go#L102-L104

**Comment by levkohimins** [comment]
@yhakbar, thanks for the review!

**Pull request merged**